src/core/lb/LBWalberla.cpp

```diff
@@ -112,6 +114,30 @@ void LBWalberla::sanity_checks(System::System const &system) const {
                              system.get_time_step());
 }
 
+void LBWalberla::on_lees_edwards_change() {
+  auto &system = System::get_system();
+  auto const energy_conversion =
+      Utils::int_pow<2>(lb_params->get_agrid() / lb_params->get_tau());
```

Comment:
Out of curiosity, would there be a way to move the LE collision operator logic out of this event into a self-contained function? Because then we could call it from the event and also from the script interface, which would greatly reduce code duplication. We could call this function `set_collision_model()` and have two overloads, one for LE and one for thermalized. This would reflect the API of `LBWalberlaBase` in the core. This would also remove the need for the script interface to query the system global variable, and least in the particular case of setting the collision operator.

A possible implementation could look like this:
```c++
struct LBWalberla {
  void set_collision_model(std::unique_ptr<LeesEdwardsPack> &&lees_edwards_pack);
  void set_collision_model(double kT, unsigned int seed);
  void update_collision_model() {
    auto const &system = System::get_system();
    if (auto le_protocol = system.lees_edwards->get_protocol()) {
      // call Lees-Edwards overload
    } else {
      // call thermalized overload
    }
  }
  void LBWalberla::on_lees_edwards_change() {
    update_collision_model();
  }
};
```